File: backend/app/schemas/review.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, field_validator


class ReviewCreate(BaseModel):
    order_id: UUID
    rating: int
    content: str | None = None
    images: list[str] | None = None

    @field_validator("rating")
    @classmethod
    def validate_rating(cls, v: int) -> int:
        if v < 1 or v > 5:
            raise ValueError("评分须为1-5")
        return v


class ReviewResponse(BaseModel):
    id: UUID
    order_id: UUID
    device_id: UUID
    reviewer_id: UUID
    rating: int
    content: str | None = None
    images: list[str] | None = None
    created_at: datetime
    updated_at: datetime
    reviewer_nickname: str | None = None
    reviewer_avatar: str | None = None

    model_config = {"from_attributes": True}


class ReviewStatsResponse(BaseModel):
    avg_rating: float
    total_reviews: int
    rating_distribution: dict[str, int]

File: frontend/src/api/review.ts
import request from '../utils/request'

export interface ReviewCreate {
  order_id: string
  rating: number
  content?: string
  images?: string[]
}

export interface ReviewItem {
  id: string
  order_id: string
  device_id: string
  reviewer_id: string
  rating: number
  content: string | null
  images: string[] | null
  created_at: string
  updated_at: string
  reviewer_nickname: string | null
  reviewer_avatar: string | null
}

export interface ReviewStats {
  avg_rating: number
  total_reviews: number
  rating_distribution: Record<string, number>
}

export function createReview(data: ReviewCreate) {
  return request.post('/reviews', data)
}

export function getDeviceReviews(deviceId: string, params?: { page?: number; size?: number }) {
  return request.get(`/devices/${deviceId}/reviews`, { params })
}

export function getUserReviewStats(userId: string) {
  return request.get(`/users/${userId}/review-stats`)
}

export function uploadFile(file: File) {
  const formData = new FormData()
  formData.append('file', file)
  return request.post('/upload', formData, { headers: { 'Content-Type': 'multipart/form-data' } })
}